### PR TITLE
Move External Functions to libffi Extension and Make Cross-Platform

### PR DESCRIPTION
**Pull request by time-killer-games**
Demo EXE and Example GM81: [DialogModule.zip](https://github.com/enigma-dev/enigma-dev/files/4537417/DialogModule.zip)

### CODE REVIEW

**Comment by time-killer-games** [comment]
@JoshDreamland ready for review me boi

File: ENIGMAsystem/SHELL/Universal_System/Extensions/ExternalFuncs/About.ey [hunk 1]
```diff
@@ -0,0 +1,13 @@
+%e-yaml
+---
+
+Name: External Functions
+Identifier: ExternalFuncs
+Author: Josh Ventura
+Description: Enables the use of GameMaker's legacy external functions for calling stuff from various DLL's. Requires the libffi dependency.
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/ExternalFuncs/About.ey:7]: I wouldn't call these legacy functions, which has a negative connotation. I usually use the word classic, but in this case I wouldn't even say that, they are still in GMS and they very clearly have a purpose and a use case.

File: ENIGMAsystem/SHELL/Universal_System/Extensions/ExternalFuncs/POSIX/ExternalFuncs.h [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#include <dlfcn.h>
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/ExternalFuncs/POSIX/ExternalFuncs.h:1]: The files `POSIX/ExternalFuncs.h` and `Win32/ExternalFuncs.h` can be combined if you just move these includes into the sources. Please do this.

File: ENIGMAsystem/SHELL/Universal_System/Extensions/ExternalFuncs/ExternalFuncs.h [hunk 3]
```diff
@@ -0,0 +1,3 @@
+void *ExternalLoad(const char *dll);
+bool ExternalFree(void *handle);
+void *ExternalFunc(void *handle, const char *func);
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/ExternalFuncs/ExternalFuncs.h:3]: Move these to namespace ENIGMA since they are internal only.

**Pull request merged**